espoon-voltti/hoivakoti-app
# feat/HOIV-128-display-all-nursinghome-data

#### Summary
Show all nursing home data on the update page so that representatives are able to modify information themselves easily.

- PageUpdate.tsx: added different inputs, types, validation and functionality for updating the data. Moved ImageUpload component to its own file ImageUpload.tsx for readability.
- PageUpdateImages.tsx: Created new sub page for changing nursing home images. The template is same as before.
- Checkbox.tsx: Added optional onBlur property.
- Radio.tsx: Added optional onBlur and value properties.
- controllers.ts, models.tx, routes.ts: Added new route and functionality to updating nursing home information in the database.
- Added new translations

#### Dependencies
No dependencies.

#### Testing instructions

On development server navigate to nursing home update page. You'll need an update key for this, you can easily log it from the database for example. Test changing/adding new values to form fields and saving the data. Remember also to test the validation, by leaving some fields empty and trying to submit. Data changes should appear on the nursing home page immediately.
#### Checklist for pull request creator


- [x] A task has been created for the PR on the Kanban board with necessary details filled (one task / repo)+t#Commit-viestien-muotoilu)
- [ ] Tests have been written for the change
- [x] All tests pass (unit, e2e)
- [x] All new code has been linted and there aren't any lint errors
- [x] The change has been tested locally, e.g., with httpie
- [x] The change has been tested in the browser with the latest Chrome
- [x] The change has been tested with a smaller screen (tablet and smartphone / emulator)
- [ ] All translations have been added (fi and sv)
- [x] The code is self-documenting or has been documented sufficiently, e.g., in the README or inline
- [x] The branch has been rebased against dev branch before the PR was created
